src/VisualStudio/IntegrationTest/TestUtilities/OutOfProcess/GenerateTypeDialog_OutOfProc.cs

```diff
@@ -0,0 +1,103 @@
+
+using System;
+using System.Linq;
+using System.Windows.Automation;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess;
+
+namespace Microsoft.VisualStudio.IntegrationTest.Utilities.OutOfProcess
+{
+    public class GenerateTypeDialog_OutOfProc : OutOfProcComponent
+    {
+        private const string GenerateTypeDialogID = "GenerateTypeDialog";
+
+        public GenerateTypeDialog_OutOfProc(VisualStudioInstance visualStudioInstance)
+            : base(visualStudioInstance)
+        {
+        }
+
+        public void VerifyOpen()
+        {
+            var dialog = DialogHelpers.FindDialog(GetMainWindowHWnd(), GenerateTypeDialogID, isOpen: true);
+
+            if (dialog == null)
+            {
+                throw new InvalidOperationException($"Expected the '{GenerateTypeDialogID}' dialog to be open but it is not.");
+            }
+        }
+
+        public void VerifyClosed()
+        {
+            var dialog = DialogHelpers.FindDialog(GetMainWindowHWnd(), GenerateTypeDialogID, isOpen: false);
+
+            if (dialog != null)
+            {
+                throw new InvalidOperationException($"Expected the '{GenerateTypeDialogID}' dialog to be closed but it is not.");
+            }
+        }
+
+        public void SetAccessibility(string accessibility)
+        {
+            DialogHelpers.SelectComboBoxItem(GetMainWindowHWnd(), GenerateTypeDialogID, "AccessList", accessibility);
+        }
+
+        public void SetKind(string kind)
+        {
+            DialogHelpers.SelectComboBoxItem(GetMainWindowHWnd(), GenerateTypeDialogID, "KindList", kind);
+        }
+
+        public void SetTargetProject(string projectName)
+        {
+            DialogHelpers.SelectComboBoxItem(GetMainWindowHWnd(), GenerateTypeDialogID, "ProjectList", projectName);
+        }
+
+        public void SetTargetFileToNewName(string newFileName)
+        {
+            DialogHelpers.SelectRadioButton(GetMainWindowHWnd(), GenerateTypeDialogID, "CreateNewFileRadioButton");
+            DialogHelpers.SetElementValue(GetMainWindowHWnd(), GenerateTypeDialogID, "CreateNewFileComboBox", newFileName);
+        }
+
+        public void SetTargetFileToExisting(string existingFileName)
+        {
+            DialogHelpers.SelectRadioButton(GetMainWindowHWnd(), GenerateTypeDialogID, "AddToExistingFileRadioButton");
+            DialogHelpers.SetElementValue(GetMainWindowHWnd(), GenerateTypeDialogID, "AddToExistingFileComboBox", existingFileName);
+        }
+
+        /// <summary>
+        /// Clicks the "OK" button and waits for the related Code Action to complete.
+        /// </summary>
+        public void ClickOK()
+        {
+            DialogHelpers.PressButtonWithName(GetMainWindowHWnd(), GenerateTypeDialogID, "OK");
+            VisualStudioInstance.VisualStudioWorkspace.WaitForAsyncOperations(FeatureAttribute.LightBulb);
```

Comment:
That's how you wait for the Generate Type operation to complete, since it was kicked off by a lightbulb.

Every Generate Type Dialog test will need to wait for the operation to complete before checking the state of the text in the editor, so it's better to put it here instead of forcing each individual test to do that (and possibly forgetting to do it, leading to flakiness).